docs/introduction-to-smart-contracts.rst

```diff
@@ -600,3 +600,8 @@ precompiled contracts. It might also be possible that new
 precompiled contracts are added to the Ethereum main chain in the future,
 but you can reasonably expect them to always be in the range between
 ``1`` and ``0xffff`` (inclusive).
+
+.. warning::
+    Since the v[0.8.10](https://github.com/ethereum/solidity/commit/a1aa9d2d90f2f7e7390408e9005d62c7159d4bd4) release the compiler does not check `extcodesize` on external calls if return data is expected, since an empty code will be unable to return data, and the ABI decoder will revert. This is an exception for external calls over precompiled contracts which are able to return data without code associated with their addresses.
```

Comment:
Some remarks on our docs conventions and syntax:
- Note that this file is RestructuredText, not Markdown. Some syntax differs, like links or double backticks.
- We don't normally link to the release in those notes. Not sure it adds anything, especially if it's just pointing at the commit. If there was a good reason to do it, I'd link to the release post on the blog.
- Please break long lines. Best after each sentence.

```
Since the 0.8.10 release the compiler does not check ``extcodesize`` on external calls if return data is expected,
since an empty code will be unable to return data, and the ABI decoder will revert.
This is an exception for external calls over precompiled contracts which are able to return data without code associated with their addresses.
```